From b5e11f9882f9137ef193a5acb02f7ee562513488 Mon Sep 17 00:00:00 2001
From: Sathya Perla <sathya.perla@emulex.com>
Date: Tue, 12 Feb 2013 23:05:19 +0000
Subject: [PATCH 804/858] be2net: remove BUG_ON() in be_mcc_compl_is_new()

The current code expects that the last word (with valid bit)
of an MCC compl is DMAed in one shot. This may not be the case.
Remove this assertion.

Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 9e9ff4b766e25624198b623fe3eda0b3e6c669cd)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/emulex/benet/be_cmds.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/emulex/benet/be_cmds.c b/drivers/net/ethernet/emulex/benet/be_cmds.c
index 8b04880..071aea7 100644
--- a/drivers/net/ethernet/emulex/benet/be_cmds.c
+++ b/drivers/net/ethernet/emulex/benet/be_cmds.c
@@ -93,13 +93,16 @@ static void be_mcc_notify(struct be_adapter *adapter)
  * little endian) */
 static inline bool be_mcc_compl_is_new(struct be_mcc_compl *compl)
 {
+ u32 flags;
+
  if (compl->flags != 0) {
-  compl->flags = le32_to_cpu(compl->flags);
-  BUG_ON((compl->flags & CQE_FLAGS_VALID_MASK) == 0);
-  return true;
- } else {
-  return false;
+  flags = le32_to_cpu(compl->flags);
+  if (flags & CQE_FLAGS_VALID_MASK) {
+   compl->flags = flags;
+   return true;
+  }
  }
+ return false;
 }
 
 /* Need to reset the entire word that houses the valid bit */
-- 
1.8.3.1